libs/ledger/src/consensus/stake_manager.cpp

```diff
@@ -16,70 +16,123 @@
 //
 //------------------------------------------------------------------------------
 
-#include "entropy/entropy_generator_interface.hpp"
 #include "ledger/chain/block.hpp"
 #include "ledger/consensus/stake_manager.hpp"
 #include "ledger/consensus/stake_snapshot.hpp"
+#include "ledger/storage_unit/storage_unit_interface.hpp"
 #include "logging/logging.hpp"
+#include "storage/resource_mapper.hpp"
 
 #include <algorithm>
 #include <cstddef>
 #include <iterator>
 
 namespace fetch {
 namespace ledger {
+namespace {
+
 constexpr char const *LOGGING_NAME = "StakeMgr";
 
-StakeManager::StakeManager(uint64_t cabinet_size)
-  : cabinet_size_{cabinet_size}
-{}
+storage::ResourceAddress const STORAGE_ADDRESS{"fetch.token.state.aggregation.stake"};
```

Comment:
This is quite a generic name for something this specialised. Rename to `STAKE_STORAGE_ADDRESS` or similar.